core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java

```diff
@@ -0,0 +1,75 @@
+package net.iakovlev.timeshape;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import java.io.BufferedOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.io.File;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+@RunWith(JUnit4.class)
+public class TimeZoneEngineSerializationTest {
+    private static TimeZoneEngine engine = TimeZoneEngine.initialize(0, 5, 0, 5, false);
+
+    @Test
+    public void testSerialzation() {
+        File f = new File ("./Engine.cache");
+        try {
+            serializeTimeZoneEngine (f, engine);
+            TimeZoneEngine engine2 = deserializeTimeZoneEngine (f);
+            f.delete ();
```

Comment:
How about we add some smoke test here after deserialization, by trying to query for some time zone that's expected to be found in the selected area? This is just to make sure deserialization has restored the state of the classes correctly. E.g. something similar to this: https://github.com/RomanIakovlev/timeshape/blob/1861cd3fd83432c80b29eeb4856718b68847228a/core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineTest.java#L23

Reply:
I've changed the loaded area to match the other test and added a check.
```
assertEquals(engine.query(52.52, 13.40), engine2.query(52.52, 13.40));
```